protocol/describeconfigs/describeconfigs.go

```diff
@@ -87,11 +88,19 @@ func (r *Response) Merge(requests []protocol.Message, results []interface{}) (
 	response := &Response{}
 
 	for _, result := range results {
-		brokerResp := result.(*Response)
-		response.Resources = append(
-			response.Resources,
-			brokerResp.Resources...,
-		)
+		switch v := result.(type) {
+		case *Response:
+			response.Resources = append(
+				response.Resources,
+				v.Resources...,
+			)
+
+		case error:
+			return nil, v
+
+		default:
+			return nil, fmt.Errorf("Unknown result type: %T", result)
```

Comment:
Looks like based on https://pkg.go.dev/github.com/segmentio/kafka-go@v0.4.31/protocol#Merger this case should trigger a panic.

Reply:
Hi @cyx, just wanted to bump this thread. Lemmie know if you don't have time to update this PR and I could take it over. Thanks!

Reply:
Hi @rhansen2 yes if you have bandwidth would appreciate a fix for this indeed. We seem to be hitting a lot of these in production.

Reply:
As for your assertion, I'm not sure about that -- what we're getting is a `*net.OpError` which goes to line 98 error case. If you meant that the default should trigger a panic, sounds good to me.